### PR TITLE
fix: add loading feedback when creating contributor insight

### LINKED ISSUE #3909: Bug: Creating or updating a contributor insight provides no feedback to user making it look broken.
The Create Insight and Update Insight buttons provide no feedback that they're updating or creating a contributor insight. It's not as noticeable with a few people being added, but if there are more than a dozen, it's noticeable.

CleanShot 2024-08-09 at 18 10 21

It needs the loading spinner to provide feedback like when we create or update a workspace.

https://github.com/open-sauced/app/blob/b1e1e203dd880054007853a9c1be75095f8e0862/pages/workspaces/new.tsx#L118-L126

https://github.com/open-sauced/app/blob/b1e1e203dd880054007853a9c1be75095f8e0862/pages/workspaces/%5BworkspaceId%5D/settings.tsx#L225-L233

### PR DESCRIPTION
**Pull request by Dun-sin**
## Description
- adds loading and loading text attribute to button component for creating insight

## Related Tickets & Documents
fixes #3909 

## Mobile & Desktop Screenshots/Recordings

https://github.com/user-attachments/assets/83c38938-520f-4743-8936-c07512e9a04e

## Steps to QA

1. Go to the create contributor insight page
2. fill in the necessary info
3. click on `create insight`

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: pages/workspaces/[workspaceId]/contributor-insights/new.tsx [hunk 1]
```diff
@@ -147,6 +147,8 @@ export default function CreateContributorInsightPage({
               variant="primary"
               className="flex gap-2.5 items-center cursor-pointer w-min mt-2 sm:mt-0 self-end"
               disabled={loading}
+              loading={loading}
+              loadingText={"Creating Insight.."}
```
Comment (nickytonline) [inline pages/workspaces/[workspaceId]/contributor-insights/new.tsx:151]: refactor (blocking): No need to appen `..`. Also we lowercase the second and later words in buttons.

```suggestion
              loadingText={"Creating insight"}
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This looks great @Dun-sin! Can you update the edit contributor insight page with the same logic as well, but with `loadingText={"Updating insight"}`?

File: pages/workspaces/[workspaceId]/contributor-insights/new.tsx (hunk 1 above)
Comment (Dun-sin) [reply pages/workspaces/[workspaceId]/contributor-insights/new.tsx:151]: got it, thanks

**Review by brandonroberts — APPROVED** [review]

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @Dun-sin! :shipit:

**Pull request merged**